src/NServiceBus.Core/Recoverability/SecondLevelRetries/SecondLevelRetries.cs

```diff
@@ -29,9 +40,10 @@ protected internal override void Setup(FeatureConfigurationContext context)
             var retryPolicy = GetRetryPolicy(context.Settings);
 
             context.Container.RegisterSingleton(typeof(SecondLevelRetryPolicy), retryPolicy);
+
             context.Pipeline.Register<SecondLevelRetriesBehavior.Registration>();
 
-            context.Container.ConfigureComponent(b => new SecondLevelRetriesBehavior(retryPolicy, context.Settings.LocalAddress()), DependencyLifecycle.InstancePerCall);
+            context.Container.ConfigureComponent(b => new SecondLevelRetriesBehavior(retryPolicy, context.Settings.LocalAddress(), b.Build<FailureInfoStorage>()), DependencyLifecycle.InstancePerCall);
```

Comment:
I tend to agree with @tmasternak , relying on a dependency from a different feature seems to be a brittle solution. Even if we do, this feature should at least depend on the `MoveFaultsToErrorQueue` feature to show the dependency.